Calculator/Calculator/Model/Formula.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  Formula.swift
+//  Calculator
+//
+//  Created by Dasan on 2023/06/02.
+//
+
+struct Formula {
+    var operands: CalculatorItemQueue<Double>
+    var operators: CalculatorItemQueue<Operator>
+
+    mutating func result() throws -> Double {
+        guard var resultValue = operands.dequeue() else {
+            throw CalculatorError.FailedOperation
+        }
+        
+        while operators.isEmpty == false {
+            guard let rightValue = operands.dequeue(),
+                  let currentOperator = operators.dequeue() else {
+                return resultValue
+            }
+            resultValue = try currentOperator.calculate(lhs: resultValue, rhs: rightValue)
+        }
```

Comment:
while문으로 optional binding을 해도 될 것 같아요.
[Optional Binding](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/thebasics/#Optional-Binding)을 보면 
> Optional binding can be used with if and while statements

라고 나와있습니다. 😆
그렇다면 아래처럼 코드를 변경해볼 수 있겠네요.
어떻게 생각하시나요?
```suggestion
        while let rightValue = operands.dequeue(), let currentOperator = operators.dequeue() {
            resultValue = try currentOperator.calculate(lhs: resultValue, rhs: rightValue)
        }
```

Reply:
중복되는 코드를 줄일 수 있는 좋은 방법이라 생각하여 `while let문`으로 수정하였습니다😆